pkg/controller/migmigration/migmigration_controller.go

```diff
@@ -318,3 +325,22 @@ func (r *ReconcileMigMigration) setOwnerReference(migration *migapi.MigMigration
 
 	return nil
 }
+
+// Ensures that the labels required to assist debugging are present on migmigration
+func (r *ReconcileMigMigration) ensureDebugLabels(migration *migapi.MigMigration) error {
+	if migration.Labels == nil {
+		migration.Labels = make(map[string]string)
+	} else {
+		if value, exists := migration.Labels[MigPlanDebugLabel]; exists {
+			if value == migration.Spec.MigPlanRef.Name {
+				return nil
+			}
+		}
+	}
+	migration.Labels[MigPlanDebugLabel] = migration.Spec.MigPlanRef.Name
+	err := r.Update(context.TODO(), migration)
```

Comment:
@jortel can you think of any complications that may arise as a result of running an `Update` on the object we're reconciling before we arrive at validations? Want to make sure this won't throw off anything with observedVersion etc.